13/abidmunirmalik/directors.py

```diff
@@ -0,0 +1,82 @@
+import csv
+from collections import defaultdict, namedtuple, Counter
+
+MOVIE_DATA = 'movie_metadata.csv'
+NUM_TOP_DIRECTORS = 20
+MIN_MOVIES = 4
+MIN_YEAR = 1960
+
+def get_movies_by_director():
+    '''Extracts all movies from csv and stores them in a dictionary
+    where keys are directors, and values is a list of movies (named tuples)'''
+    Movies  = namedtuple('Movies', ['title','year','score'])
+    movies_by_director = defaultdict(list)
+    with open(MOVIE_DATA) as csvfile:
+        reader = csv.DictReader(csvfile)
+        for rows in reader:
+            try:
+                director = rows['director_name']
+                title = rows['movie_title'].replace('\xa0','')
+                year =  int(rows['title_year'])
+                score = float(rows['imdb_score'])
+            except ValueError:
+                continue
+            movie = Movies(title=title, year=year, score=score)
+            movies_by_director[director].append(movie)     
+    return movies_by_director
+
+def filter_directors(directors):
+    directors_with_less_than_min_movies = []
+    for director,movies in directors.items():
+        d,m = director,len(movies)
+        if m < MIN_MOVIES:
+            directors_with_less_than_min_movies.append(d)
+    for _ in directors_with_less_than_min_movies:
+        del directors[_]
```

Comment:
OK I understand now that you first build a list `directors_with_less_than_min_movies` to use it for deleting all entries in the original `directors` dictionary. That is a little more complicated than it has to be. You can use list comprehensions and create tuples, and the `dict` function will give you a dictionary. So 
`return dict([(director, movies) for director, movies in directors.items() if len(movies) < MIN_MOVIES])` is all you need in that function and gives you a dictionary with only directors that have less than MIN_MOVIES